jupyterhub_singleuser_profiles/service.py

```diff
@@ -37,83 +39,54 @@ def __init__(self, server_url, token, namespace=None, verify_ssl=True):
+      confm = response.to_dict()
     except Exception as e:
       _LOGGER.error("Error: %s %s" % (name, e))
       template = None
+    template = yaml.load(confm['data'].get(path))
 
     return template
 
-  def process_template(self, user, template, **parameters):
-
-    self._set_template_parameters(
-      template,
-      USERNAME=escape(user),
-      **parameters
-    )
-    template["apiVersion"] = "template.openshift.io/v1"
-
-    endpoint = "{}/apis/template.openshift.io/v1/namespaces/{}/processedtemplates".format(
-        self.server_url,
-        self.namespace
-    )
-    response = requests.post(
-        endpoint,
-        json=template,
-        headers={
-            'Authorization': 'Bearer {}'.format(self.token),
-            'Content-Type': 'application/json'
-        },
-        verify=False #FIXME?
-    )
-    _LOGGER.debug("OpenShift master response template (%d): %r", response.status_code, response.text)
+  def process_template(self, user, service_name, template, configuration, labels=None):
 
-    try:
-        response.raise_for_status()
-    except Exception:
-        _LOGGER.error("Failed to process template: %s", response.text)
-        raise
+    tmp = jinja2.Template(json.dumps(template))
 
-    processed = response.json()
-    if len(processed["objects"]) != 1:
-      raise Exception("Template must contain a single ConfigMap or CustomResource")
+    result = tmp.render(parameters=configuration)
 
-    configMap = processed["objects"][0]
-    if not configMap["metadata"].get("labels"):
-      configMap["metadata"]["labels"] = {}
+    result = json.loads(result)
+    if not result.get('metadata'):
+      result['metadata'] = {}
+    if labels:
+      result['metadata']['labels'] = labels
+    
+    result['metadata']['name'] = service_name + '-' + user
 
-    configMap["metadata"]["labels"][_SERVICE_LABEL] = escape(user)
+    return result
 
-    return configMap
 
-  def submit_resource(self, resource, return_paths):
-    if resource["kind"] == "ConfigMap":
-      name = resource["metadata"]["name"]
-      try: 
-        api_response = self.k8s_api_instance.replace_namespaced_config_map(name, self.namespace, resource, pretty="true", )
-        result = self._get_data_from_response(api_response, return_paths)
+  def submit_resource(self, processed_template, return_paths):
+    client_wrapper = self.os_client.resources.get(api_version=processed_template['apiVersion'], kind=processed_template['kind'])
+    try:
+      response = client_wrapper.create(body=processed_template, namespace=self.namespace)
+      result = self._get_data_from_response(response, return_paths)
+      return result
+    except ApiException as e:
+      try:
+        original_resource = client_wrapper.get(name=processed_template['metadata'].get('name'), namespace=self.namespace)
+        original_dict = original_resource.to_dict()
+        processed_template['metadata']['resourceVersion'] = original_dict['metadata']['resourceVersion']
```

Comment:
Can you please add a comment pointing to https://github.com/kubernetes/kubernetes/issues/70674#issuecomment-438569688, so that we know why we need this line?